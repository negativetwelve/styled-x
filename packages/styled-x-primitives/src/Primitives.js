// Libraries
import {Image, Text, Touchable, View} from 'react-x-primitives';
import createComponent from 'styled-x-create-component';

// Inject Primitives to Styled that we can use `.extend` on.
const Primitives = {
  Image: createComponent(Image),
  Text: createComponent(Text),
  Touchable: createComponent(Touchable, {
    propTypes: {...Touchable.propTypes, ...View.propTypes},
  }),
  View: createComponent(View),
};

// This allows us to import the entire Primitives object or specific
// components via `import {View} from 'styled-x-primitives';`.
module.exports = Primitives;
