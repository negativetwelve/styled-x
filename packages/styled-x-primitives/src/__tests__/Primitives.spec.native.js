// Modules
import React from 'react';
import Primitives from '../Primitives';

const Title = Primitives.Text.extend`
  font-size: 16px;
  text-align: center;
`;

/* eslint-disable no-undef */
describe('Primitives`', () => {
  describe('Title', () => {
    itRenders(() => <Title />);
  });
});
