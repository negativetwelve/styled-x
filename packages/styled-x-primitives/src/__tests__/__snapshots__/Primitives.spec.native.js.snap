// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primitives\` Text Title renders snapshot 1`] = `
<div
  className="css-text-901oao r-fontSize-ubezar r-textAlign-q4m81j"
  dir="auto"
/>
`;

exports[`Primitives\` Text TitleWithProps renders snapshot 1`] = `
<div
  className="css-text-901oao r-color-jwli3a r-fontSize-adyw6z"
  color="#FFFFFF"
  dir="auto"
  size={20}
/>
`;

exports[`Primitives\` View Square renders snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-height-z80fyv r-width-19wmn03"
/>
`;

exports[`Primitives\` View SquareWithProps renders snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-borderRadius-1f0042m r-height-z80fyv r-width-19wmn03"
/>
`;
